### ISSUE TITLE
Navigation - Configurable keyboard and gamepad controls

### ISSUE THREAD

**Issue by notskm** [ocornut/imgui#6506]
**Version/Branch of Dear ImGui:**

Version: 1.89.6
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl2.cpp + imgui_impl_sdlrenderer2.cpp
Compiler: MSVC, clang, gcc
Operating System: Windows 10, Ubuntu

**My Issue/Question:**

It is currently possible to navigate using a gamepad or a keyboard. As far as I can tell, there is no way to rebind the buttons/keys to specific actions. For example, the X button on an Xbox controller opens the menu to navigate between windows. Personally, I believe the "view button" (the button with the two squares) makes more sense for my use-case. As for the keyboard, we currently navigate between GUI elements using the arrow keys. It would be nice if we could navigate rebind this action to WASD instead.

To sum up, all keybinds and gamepad controls should be configurable.

**Comment by ocornut** [comment]
In the earlier version we had an indirection (in the form of named `ImGuiNavInput_` enum) which was initially intended to do that. Over time I realized it was too difficult to full-fill, there are too many subtleties in nav controls, and user application may need to rely on some of that knowledge. I decided to remove that indirection and things became simpler. 

I don't intend to provide full rebind but improvements are possible. For example using back/view to open a modal version of the ctrl+tab switcher may be possible if it was opt-in. But I would encourage you to try get accustomed to those controls.
Depending on exact needs and your user base you might be able to merge WASD keys into Arrow keys (using a technique similar to this post https://github.com/ocornut/imgui/pull/5723#issuecomment-1260905083).

There is a distinct need to leave at least one button unused in default setup to facilitate user app providing a way to "return" back to game/app controls.
